outline.md

```diff
@@ -0,0 +1,292 @@
+- I’m using an international keyboard and keys that use AltGr or Ctrl+Alt aren’t working
+- I’m having a problem with Julia! What do I do?
+- I’m getting an error about a “self-signed certificate”. What do I do?
+- I’m having a problem with PlatformIO! What do I do?
+- How do I make Atom recognize a file with extension X as language Y?
+- How do I make the Welcome screen stop showing up?
+- How do I preview web page changes automatically?
+- How do I accept input from my program or script when using the script package?
+- I am unable to update to the latest version of Atom on macOS. How do I fix this?
+- I’m trying to change my syntax colors from styles.less, but it isn’t working!
+- How do I build or execute code I've written in Atom?
+- How do I uninstall Atom on macOS?
+- macOS Mojave font rendering change
+- Why does macOS say that Atom wants to access my calendar, contacts, photos, etc.?
+- How do I turn on line wrap?
+- The menu bar disappeared, how do I get it back?
+- How do I use a newline in the result of find and replace?
+- What is this line on the right in the editor view?
+
+### Appendix C: Shadow DOM
+
+- Removing Shadow DOM styles
+
+### Appendix D: Upgrading to 1.0 APIs
+
+- Upgrading Your Package
+- Upgrading Your UI Theme Or Package Selectors
+- Upgrading Your Syntax Theme
+
+### Appendix E: Atom server-side APIs
+
+- Atom package server API
+- Atom update server API
+
+## The remaining bits to be written and finished:
+
+### Chapter 3: Hacking Atom
+
+- Tutorials
+  - Word Count Package (Basic)
+  - Word Count Status Bar (status bar manipulation)
+  - Go To Line Package (simple with feedback and cursor movement)
+  - Open On GitHub - (no UI, just run command)
+  - Snippets - (modify text in a buffer)
+  - Linter - (modify text in a buffer, shell to external command)
+  - Preview Pane (add a panel)
+  - Fuzzy Finder (modal and open file in new tab)
+  - Tree View / Mini-map (add a sidebar panel)
+  - Status Bar (add small panel)
+  - Git-Log / Settings View (standalone UI in tab)
+  - Git-status / Bookmarks (decorate gutter)
+  - Autocomplete / Color picker (in-context UI)
+  - Vim-mode (editor remapping)
+  - Image View (file type rendering handler)
+  - Language (GFM?)
+    - Creating grammars
+    - converting-a-text-mate-bundle.md
+  - Unity UI (UI theme)
+    - converting-a-text-mate-theme.md
+    - theme-variables.md
+    - Shadow DOM
+    - https://atom.io/docs/latest/upgrading/upgrading-your-ui-theme
+
+## Chapter 4: Behind Atom
+
+- advanced dev tools
+  - keybinding resolver
+  - timecop
+- Maintaining Packages
+  - how to publish
+    - Atom.io package system (adding, searching, stars)
+  - changelog
+  - issue tracker
+- Best practices (patterns and anti-patterns)
+  - aggregating subscriptions
+  - view best practices
+  - don't make a slow package
+  - CI for packages (https://blog.atom.io/2014/04/25/ci-for-your-packages.html)
+
+### Include in Chapter 3 or 4
+
+- Make sure we include (in Ch 3 or Ch4 or both)
```

Comment:
```suggestion
- Make sure we include (in Ch 3 or Ch4 or both):
```

Reply:
Not adding this one, this is a section we will be reorganizing, and the chapter headings will be disappearing. This section was where they weren't sure where exactly to add them.